.github/workflows/checks.yml

```
@@ -53,3 +55,63 @@ jobs:
       run: yarn coverage
 
     - uses: codecov/codecov-action@v5
+
+  update-foundry-submodule-docs:
+    if: github.event_name == 'repository_dispatch'
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        submodules: true 
+        fetch-depth: 0
+
+    - name: Set up environment
+      uses: ./.github/actions/setup
+
+    - name: Update submodule
+      run: |
+        cd submodules/openzeppelin-foundry-upgrades
+        git fetch origin main
+        git reset --hard origin/main
+
+    - name: Regenerate docs
+      run: yarn docs:foundry:ci
+
+    - name: Check for changes in docs/modules/ROOT/pages/foundry/
+      id: check_changes
+      run: |
+        [ -n "$(git diff --name-only HEAD -- docs/modules/ROOT/pages/foundry/)" ] && echo "SHOULD_COMMIT_DOC_UPDATE=true" >> $GITHUB_ENV || true
+
+    - name: Set Git identity
+      if: env.SHOULD_COMMIT_DOC_UPDATE == 'true'
+      run: |
+        git config --global user.name "GitHub Actions"
+        git config --global user.email "github-actions@github.com"
```

According to [actions/checkout@v4 docs](https://github.com/actions/checkout/tree/v4?tab=readme-ov-file#push-a-commit-using-the-built-in-token), we should use a specific user name/email format for the GitHub Actions bot.  Is there a need to use `--global` here?

```suggestion
        git config user.name "github-actions[bot]"
        git config user.email "41898282+github-actions[bot]@users.noreply.github.com"
```